clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,9 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+      if (!LastArg)
+        LastArg = &(dyn_cast<InputAction>(UnbundlerInputs.back())->getInputArg());
+ 
```

Comment:
Please create a test that exercises the pathway in which we know LastArg will not be set before the partial link occurs.